src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.ts

```diff
@@ -0,0 +1,162 @@
+    row: number;
+    fieldRows: FieldRow[];
+    addRowSelected = 'collapsed';
+    selectedCol = 0;
+    disabledButton = false;
+    comboHotKeys: Hotkey;
+    @Output() selectedColums: EventEmitter<number> = new EventEmitter<number>();
+    @ViewChild('rowList') rowList: ElementRef;
+
+    constructor(private hotkeysService: HotkeysService , public messageService: MessageService) {
+        super(
+            [
+                'contenttypes.content.add_rows'
+            ],
+            messageService
+        );
+    }
+
+    ngOnInit(): void {
+        this.comboHotKeys = new Hotkey(['left', 'right', 'enter', 'esc'], (event: KeyboardEvent, combo: string): boolean => {
+            switch (combo) {
+                case 'left':
+                    this.leftKeyboardSelect();
+                    break;
+                case 'right':
+                    this.rightKeyboardSelect();
+                    break;
+                case 'enter':
+                    this.row = this.setRowListChild(this.selectedCol).children.length;
+                    this.selectRow(this.row);
+                    break;
+                case 'esc':
+                    this.resetRow();
+                    break;
+            }
+            return false;
+        });
+
+        this.hotkeysService.add(this.comboHotKeys);
+    }
+
+    ngOnDestroy(): void {
+        this.hotkeysService.remove(this.comboHotKeys);
+    }
+
+    /**
+     * Set columns active when mouse enter
+     * @param col
+     */
+    public mouseEnter(col: number) {
+        this.selectedCol = col;
+    }
+
+    /**
+     * Emit number of columns after select colum the reset row
+     * @param {number} row
+     * @memberof ContentTypeFieldsAddRowComponent
+     */
+    public selectRow(row: number): void {
+        this.selectedColums.emit(row);
+        this.resetRow();
+    }
+
+    /**
+     * Return columns inside each li element
+     * @param {number} n
+     * @returns {number[]}
+     * @memberof ContentTypeFieldsAddRowComponent
+     */
+    public numberOfCols(n: number): number[] {
+        return Array(n).fill('');
+    }
+
+    /**
+     * Display row when click on add rows button
+     * @memberof ContentTypeFieldsAddRowComponent
+     */
+    public addRow(): void {
+        this.addRowSelected = 'expanded';
+    }
+
+    private leftKeyboardSelect(): any {
+        this.selectedCol = this.selectedCol - 1;
+        if (this.selectedCol < 0) {
+            this.selectedCol = 3;
+        }
+    }
+
+    private rightKeyboardSelect(): any {
+        this.selectedCol = this.selectedCol + 1;
+        if (this.selectedCol > 3) {
+            this.selectedCol = 0;
+        }
+    }
+
+    private setRowListChild(col: number): any {
+        return this.rowList.nativeElement.children[col];
+    }
+
+    private resetRow(): any {
+        this.selectedCol = 0;
+        this.addRowSelected = 'collapsed';
+    }
+
+    private setTooltipValue(col: number): string {
```

Comment:
this kind of things could be handle easier to achieve with just an array associated to the column

const toolTipValueMap = new Array( 'contenttypes.content.one_column', ''contenttypes.content.two_columns'' ...);

the method will be return this.i18nMessages[this.toolTipValueMap[col]];